docs/xss.rst

```diff
@@ -0,0 +1,96 @@
+Performing XSS Vulnerability Audits
+===================================
+
+The bok-choy framework includes the ability to perform XSS (cross-site scripting) audits on
+web pages using a short XSS locator defined in
+https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#XSS_Locator.
+
+Trigger XSS Vulnerability Audits in Existing Tests
+--------------------------------------------------
+
+You might already have some bok-choy tests written for your web application. To
+leverage existing bok-choy tests and have them fail on finding XSS vulnerabilities,
+follow these steps.
+
+1. Insert the ``XSS_INJECTION`` string defined in ``bok_choy.page_object`` into your page content.
+2. Set the ``VERIFY_XSS`` environment variable to ``True``.
+
+::
+
+    export VERIFY_XSS=True
+
+
+An XSS audit is triggered whenever a page object's ``q`` method is called. The audit will detect
+improper escaping both in HTML and in Javascript that is embedded within HTML.
+
+If errors are found on the page, an XSSExposureError is raised.
+
+Here is an example of a bok-choy test that will implicitly check for XSS vulnerabilities.
```

Comment:
nit: I think the word "implicitly" is confusing here. It's not really implicit, as the test-writer intentionally imported XSS_INJECTION, and used it as the user_name. Maybe just leave it off.